src/tcpedit/edit_packet.c

```diff
@@ -63,7 +63,9 @@ fix_ipv4_checksums(tcpedit_t *tcpedit, struct pcap_pkthdr *pkthdr, ipv4_hdr_t *i
     
 
     /* calc the L4 checksum if we have the whole packet && not a frag or first frag */
-    if (pkthdr->caplen == pkthdr->len && (htons(ip_hdr->ip_off) & IP_OFFMASK) == 0) {
+    if (pkthdr->caplen == pkthdr->len
+            && pkthdr->len > ntohs(ip_hdr->ip_len)
+            && (htons(ip_hdr->ip_off) & IP_OFFMASK) == 0) {
```

Comment:
Agreed. I'll watch for your patch. Also please make PR to branch `appneta/tcpreplay/Feature_302_fuzz_option`